### PR TITLE
Fix a couple of bugs and libnvm on windows

### PR DESCRIPTION
**Pull request by musm**
Tests now pass 

fixes https://github.com/JuliaGPU/CUDArt.jl/issues/71

### CODE REVIEW

File: src/device.jl [hunk 1]
```diff
@@ -77,30 +77,44 @@ function devices(f::Function, devlist::Union{Integer,AbstractVector})
 end
 
 function filter_free(devlist)
-    usenvml = true
-    if Libdl.find_library(["libnvidia-ml"], []) == ""
-        warning("NVML not found, resorting to nvidia-smi")
```
Comment (musm) [inline src/device.jl]: warning is not a function in julia

File: src/device.jl [hunk 2]
```diff
@@ -77,30 +77,40 @@ function devices(f::Function, devlist::Union{Integer,AbstractVector})
...
+    nvml_libdir = is_windows() ? joinpath(ENV["ProgramFiles"], "NVIDIA Corporation", "NVSMI") : ""
+    if is_windows() && !isdir(nvml_libdir)
+        error("Could not determine Nvidia driver installation location.")
+    end
+    const libnvml = Libdl.find_library(["libnvidia-ml", "nvml"], [nvml_libdir])
```
Comment (musm) [inline src/device.jl]: should these be setup before hand or is it fine to just set them up in a function

Comment (vchuravy) [inline src/device.jl]: should probably be setup beforehand.

**Comment by vchuravy** [comment]
Tests currently fail because of #72. I won't immediately have time to look into that, but I would like to have a green(ish) CI before merging anything.

**Pull request closed without merging**